### ISSUE TITLE
Unable to load script from assets 'index.android.bundle'

### ISSUE THREAD

**Issue by pcegarra** [ueno-llc/react-native-starter#209]
Im running the command `react-native run-android` deploying in a Xioami A1 with android 9 in mac os.
The full error log is,,

```
    java.lang.RuntimeException: Unable to load script from assets 'index.android.bundle'. Make sure your bundle is packaged correctly or you're running a packager server.
        at com.facebook.react.bridge.CatalystInstanceImpl.jniLoadScriptFromAssets(Native Method)
        at com.facebook.react.bridge.CatalystInstanceImpl.loadScriptFromAssets(CatalystInstanceImpl.java:216)
        at com.facebook.react.bridge.JSBundleLoader$1.loadScript(JSBundleLoader.java:32)
        at com.facebook.react.bridge.CatalystInstanceImpl.runJSBundle(CatalystInstanceImpl.java:243)
        at com.facebook.react.ReactInstanceManager.createReactContext(ReactInstanceManager.java:1116)
        at com.facebook.react.ReactInstanceManager.access$900(ReactInstanceManager.java:117)
        at com.facebook.react.ReactInstanceManager$5.run(ReactInstanceManager.java:916)
        at java.lang.Thread.run(Thread.java:764)
```

Im trying to fix it reading from stackoverflow, because it seems a common problem

**Comment by jeremybarbet** [comment]
Is it working on emulator?

We had the same issue (afaik) on iOS and we wrote the documentation for that: https://ueno-llc.github.io/react-native-starter/#/DEBUGGING?id=troubleshooting

As you can see the documention is not complete for Android. My 2ct would be, that the bundle file doesn't exist and isn't include into the app apk.

I could take a look at it later, if you have any more informations feel free to share them here

**Comment by pcegarra** [comment]
Yes, its running fine on emulator (tested with android 6 and android 8).
And is getting fail on Xiaomi a1 (tested on android 8 and android 9).

**Comment by jeremybarbet** [comment]
Nice one, I'll update the doc later with it.

Unfortunately using `react-native bundle` will create a static version of the javascript code into the apk of the app. In short, you have to execute the command each time you change something to see the changes on your device.

Eventhough, the cause of the initial issue could be your network that doesn't allow your localhost ip. Make sure both you are on the same wifi network and you have access to your computer through your phone.

**Comment by birkir** [comment]
This is fixed in the latest version 3.0.0
